source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,224 @@
+===========================================
+Connections Survive Primary Step Down Tests
+===========================================
+
+These tests can be used to verify a driver's compliance with server discovery
+and monitoring requirements with respect to handling "not master" and
+"server shutting down" error responses from the server.
+
+.. contents::
+
+These tests apply only to replica set topologies.
+
+Tests
+-----
+
+The driver should implement the following tests:
+
+getMore Iteration
+`````````````````
+
+This test requires server version 4.2 or higher, and makes use of events
+defined in the `CMAP specification
+<https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst>`_.
+
+Perform the following operations:
+
+- Insert 5 documents into a collection.
+- Clear the connection pool to the current primary (see CMAP spec for
+  the definition of connection pool clear operation, and
+  `Why clear the pool in getMore test?`_ for rationale).
+- Start a find operation on the collection with a batch size of 2, and
+  retrieve the first batch of results.
+- Verify that a ConnectionCreated CMAP event has been published.
+- Send a ``replSetStepDown`` command to the current primary and verify that
+  the command succeeded, or follow `Server Step Down Procedure`_ below.
+- Retrieve the next batch of results from the cursor obtained in the find
+  operation, and verify that this operation succeeded.
+- Verify that no new ConnectionCreated CMAP events have been published.
```

Comment:
For that specific pool, right?

Reply:
Given @jyemin comment above that a pool is not accessible for these tests to assert on, it is not clear to me how to specify that no events should be published for the pool belonging to the server on which getMore was executed.

Reply:
The _pool_ is not accessible (as in the ConnectionPool interface defined in CMAP), but the pool _events_ should be via event listeners registered on a MongoClient.

Reply:
I already have the event assertion in the spec text, the suggestion was to also require that the event was for the pool related to the server which the getMore command was sent to. Can you suggest language which will accomplish this given Java driver's constraints on pool accessibility?

Reply:
We can just assert that _no_ new connections were created since the find command was initiated, regardless of which pool.  No other pool should have had a connection created in it either.

Reply:
This has been the original wording in this specification; @behackett  are you OK with the current language?